### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Ogerets**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -231,9 +231,45 @@ import VirgilE3Kit
...
+```swift
+ethree.restorePrivateKey(password: userPassword) { error in 
+    guard error == nil else {
+        // Error handling
+    }
+    // Private Key successfully restored and saved locally
+}
+```
+
+If you authorize users using password in your application, please do not use the same password to backup Private Key, since it's not secure. Instead, you can derive from your user password two different ones.
```
Comment (SanjoDeundiak) [inline README.md]: Since it breaks e2ee

File: Source/PublicAPI/Actions/EThree+KeyBackup.swift [hunk 2]
```diff
@@ -38,6 +38,22 @@ import VirgilSDK
 
 // MARK: - Extension with key back-up operations
 extension EThree {
+    // swiftlint:disable force_unwrapping
+    /// Derives different passwords for login and for backup from one
+    /// - Parameter password: password to derive from
+    @objc open func derivePasswords(from password: String) -> DerivedPasswords {
```
Comment (SanjoDeundiak) [inline Source/PublicAPI/Actions/EThree+KeyBackup.swift]: Why isn't it static? To initialise e3kit user mush provide jwt callback, for which user must be logged in, but to login user needs this function.

File: Source/PublicAPI/Deprecated/EThree+Encryption.swift [hunk 3]
```diff
@@ -178,62 +224,101 @@ extension EThree {
...
+    ///   - recipientKeys: result of lookupPublicKeys call recipient PublicKeys to sign and encrypt with
+    /// - Returns: decrypted Data
     /// - Important: Automatically includes self key to recipientsKeys.
     /// - Important: Requires private key in local storage
-    @objc(encryptStream:toStream:forUser:error:)
-    open func encrypt(_ stream: InputStream, to outputStream: OutputStream, for user: Card) throws {
-        try self.encrypt(stream, to: outputStream, for: [user.identity: user])
+    /// - Note: Avoid key duplication
+    @available(*, deprecated, message: "Use signThenEncrypt method instead.")
+    @objc func encrypt(data: Data, for recipientKeys: LookupResult) throws -> Data {
```
Comment (SanjoDeundiak) [inline Source/PublicAPI/Deprecated/EThree+Encryption.swift:249]: What's the point of changing signature for deprecated method? Doesn't that break obj-c backward compatibility?

File: Source/PublicAPI/Deprecated/EThree+Encryption.swift [hunk 4]
```diff
@@ -282,7 +369,8 @@ extension EThree {
         return encryptedData
     }
 
-    internal func decryptInternal(data: Data, from publicKey: VirgilPublicKey?) throws -> Data {
+    @available(*, deprecated)
+    internal func oldDecryptInternal(data: Data, from publicKey: VirgilPublicKey?) throws -> Data {
```
Comment (SanjoDeundiak) [inline Source/PublicAPI/Deprecated/EThree+Encryption.swift:373]: There is no point marking internal funcs deprecated

File: Source/PublicAPI/Encryption/EThree+AuthEncrypt.swift [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    /// - Important: Deprecated decrypt method is unable to decrypt result of this method
+    ///
+    /// - Parameters:
+    ///   - data: data to encrypt
+    ///   - user: user Card to encrypt for
+    /// - Returns: encrypted data
+    /// - Important: Automatically includes self key to recipientsKeys.
+    /// - Important: Requires private key in local storage
+    @objc(signThenEncryptData:forUser:error:)
+    open func signThenEncrypt(data: Data, for user: Card) throws -> Data {
```
Comment (SanjoDeundiak) [inline Source/PublicAPI/Encryption/EThree+AuthEncrypt.swift]: 1. Rename to authEncrypt
2. What's the point of setting explicit objc name?

**Review by SanjoDeundiak — CHANGES_REQUESTED** [review]

File: Source/PublicAPI/Deprecated/EThree+Encryption.swift (hunk 4 above)
Comment (Ogerets) [reply Source/PublicAPI/Deprecated/EThree+Encryption.swift:373]: The point is to silence warnings due to use of deprecated methods inside

File: Source/PublicAPI/Encryption/EThree+AuthEncrypt.swift (hunk 5 above)
Comment (Ogerets) [reply Source/PublicAPI/Encryption/EThree+AuthEncrypt.swift]: > What's the point of setting explicit objc name?

Conflicts of encrypt(for user: Card) and encrypt(for users: [Cards]). It has same signature in objc

File: Source/PublicAPI/Deprecated/EThree+Encryption.swift (hunk 3 above)
Comment (Ogerets) [reply Source/PublicAPI/Deprecated/EThree+Encryption.swift:249]: Signature of which function has beed changed?

File: README.md (hunk 1 above)
Comment (Ogerets) [reply README.md]: Fixed

File: Source/PublicAPI/Actions/EThree+KeyBackup.swift (hunk 2 above)
Comment (Ogerets) [reply Source/PublicAPI/Actions/EThree+KeyBackup.swift]: Fixed

File: Source/PublicAPI/Encryption/EThree+AuthEncrypt.swift (hunk 5 above)
Comment (Ogerets) [reply Source/PublicAPI/Encryption/EThree+AuthEncrypt.swift]: Renamed

**Review by SanjoDeundiak — APPROVED** [review]

**Pull request merged**